src/sentry/mediators/external_requests/issue_link_requester.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+
+import six
+import logging
+from uuid import uuid4
+
+from six.moves.urllib.parse import urlparse
+from sentry.http import safe_urlopen, safe_urlread
+from sentry.coreapi import APIError
+from sentry.mediators import Mediator, Param
+from sentry.mediators.external_requests.util import validate
+from sentry.utils import json
+from sentry.utils.cache import memoize
+
+logger = logging.getLogger('sentry.mediators.external-requests')
+
+
+class IssueLinkRequester(Mediator):
+    """
+    1. Makes a POST request to another service with data used for creating or
+    linking a Sentry issue to an issue in the other service.
+
+    The data sent to the other service is always in the following format:
+        {
+            'installtionId': <install_uuid>,
+            'issueId': <sentry_group_id>,
+            'webUrl': <sentry_group_web_url>,
+            <fields>,
+        }
+
+    <fields> are any of the 'create' or 'link' form fields (determined by
+    the schema for that particular service)
+
+    2. Validates the response format from the other service and returns the
+    payload.
+
+    The data sent to the other service is always in the following format:
+        {
+            'identifier': <some_identifier>,
+            'webUrl': <external_issue_web_url>,
+            'project': <top_level_identifier>,
+        }
+
+    The project and identifier are use to generate the display text for the linked
+    issue in the UI (i.e. <project>#<identifier>)
+    """
+
+    install = Param('sentry.models.SentryAppInstallation')
+    uri = Param(six.string_types)
+    group = Param('sentry.models.Group')
```

Comment:
Should we call this `issue` to match how we talk about or keep it the name of the model?